src/database/DataTypes/StocktakeItem.js

```diff
@@ -305,7 +305,16 @@ export class StocktakeItem extends Realm.Object {
    */
   get hasReason() {
     if (!this.batches) return false;
-    return this.batches.some(batch => batch.option);
+    return this.batches.every(batch => {
+      if (batch.snapshotTotalQuantity !== batch.countedTotalQuantity) {
+        return !!batch.option;
+      }
+      return true;
+    });
```

Comment:
I feel like the getter name and description don't match this anymore.

I wonder if it's useful semantically to not have these as getters. Idk: